ci/scripts/cpp_build.sh

```diff
@@ -118,12 +118,36 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
     fi
   }
 
+  ORIGINAL_CC="${CC}"
+  if [ -n "${CC}" ]; then
+    if [ "${ARROW_USE_CCACHE}" = "ON" ]; then
+      CC="ccache ${CC}"
+    else
+      if command -v sccache; then
```

Comment:
This will only enable `sccache` if `CC` is set, is that what we want?

Reply:
When CC is not set if Meson detects a caching tool on the system Meson should use it automatically. sccache gets a higher priority than ccache. 

See also https://mesonbuild.com/Feature-autodetection.html#ccache

Reply:
Ah, ok. It's a pity we have to replicate the same logic here...